src/veh_interact.cpp

```diff
@@ -316,30 +315,19 @@ task_reason veh_interact::cant_do (char mode)
         } );
         has_tools = true; // checked later
         break;
-    case 'f': // refill mode
-        if (!ptanks.empty()) {
-            std::vector<vehicle_part *>::iterator iter;
-            for (iter = ptanks.begin(); iter != ptanks.end(); ) {
-                if ((*iter)->hp > 0 &&
-                    g->refill_vehicle_part(*veh, *iter, true)) {
-                    pass_checks = true;
-                    iter++;
-                } else {
-                    iter = ptanks.erase(iter);
-                }
-            }
-            if(pass_checks) {
-                return CAN_DO;
-            }
-        }
 
-        // if refillable parts exist but can't be refilled for some reason.
-        if (has_ptank) {
-            return CANT_REFILL;
-        }
+    case 'f': {
+        // first find all possible ammo
+        auto opts = g->u.items_with( []( const item &e ) { return e.is_ammo(); } );
```

Comment:
> bleach  doesn't have an  ammo_type  so this won't work after  3047635 .

Exactly. A regression: something was possible before and is not possible afterwards.

> For now fuel tanks have to contain ammo; this is sufficient to both fix the bug and is not a regression from  master .

It would probably not harm to change the check to something like `e.is_ammo() || e.made_of(LIQUID)` and thereby keep the existing behavior.

> Generic vehicle tanks requires more support and testing and can be considered as part of a separate PR.

This is not about generic tanks. This is about the existing tank system.